x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/execution/search/Querier.java

```diff
@@ -144,6 +145,10 @@ public void query(List<Attribute> output, QueryContainer query, String index, Ac
             l = new ScrollActionListener(listener, client, cfg, output, query);
         }
 
+        if (cfg.task() != null && cfg.task().isCancelled()) {
```

Comment:
There are a couple of places where the task is checked against being canceled within SQL, both placed before "leaving" SQL:
* before resolving the index;
* here, before running the search.

It makes sense to do it there because (1) compared to the distributed part, the SQL code will run quick; and (2) there's anyways a race between the cancelation and task execution -- one could add more checkpoints throughout SQL, but it won't practically improve much (like reactivity to the cancelation).